From 2d6a3e38e18f334000764230e8f372f4dd43774a Mon Sep 17 00:00:00 2001
From: Yu Qiaowei <cerf.yu@rock-chips.com>
Date: Fri, 22 Jan 2021 11:03:42 +0800
Subject: [PATCH] video/rockchip: rga2: Modify the judgment of first req.

Fixed the error that color fill would cause "req argument is inval".

Signed-off-by: Yu Qiaowei <cerf.yu@rock-chips.com>
Change-Id: I94ebfc2061f16916813a0d85630f8a9862728b0e
---
 drivers/video/rockchip/rga2/rga2_drv.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/video/rockchip/rga2/rga2_drv.c b/drivers/video/rockchip/rga2/rga2_drv.c
index 50297f4d0c8f..ca2148506c35 100644
--- a/drivers/video/rockchip/rga2/rga2_drv.c
+++ b/drivers/video/rockchip/rga2/rga2_drv.c
@@ -1695,7 +1695,7 @@ static long rga_ioctl(struct file *file, uint32_t cmd, unsigned long arg)
 			}
 			RGA_MSG_2_RGA2_MSG(&req_rga, &req);
 
-			if (first_RGA2_proc == 0 && req.bitblt_mode == bitblt_mode && rga2_service.dev_mode == 1) {
+			if (first_RGA2_proc == 0 && req.render_mode == bitblt_mode && rga2_service.dev_mode == 1) {
 				memcpy(&req_first, &req, sizeof(struct rga2_req));
 				if ((req_first.src.act_w != req_first.dst.act_w)
 						|| (req_first.src.act_h != req_first.dst.act_h)) {
@@ -1721,7 +1721,7 @@ static long rga_ioctl(struct file *file, uint32_t cmd, unsigned long arg)
 			}
 
 			RGA_MSG_2_RGA2_MSG(&req_rga, &req);
-			if (first_RGA2_proc == 0 && req.bitblt_mode == bitblt_mode && rga2_service.dev_mode == 1) {
+			if (first_RGA2_proc == 0 && req.render_mode == bitblt_mode && rga2_service.dev_mode == 1) {
 				memcpy(&req_first, &req, sizeof(struct rga2_req));
 				if ((req_first.src.act_w != req_first.dst.act_w)
 						|| (req_first.src.act_h != req_first.dst.act_h)
@@ -1869,7 +1869,7 @@ static long compat_rga_ioctl(struct file *file, uint32_t cmd, unsigned long arg)
 
 			RGA_MSG_2_RGA2_MSG_32(&req_rga, &req);
 
-			if (first_RGA2_proc == 0 && req.bitblt_mode == bitblt_mode && rga2_service.dev_mode == 1) {
+			if (first_RGA2_proc == 0 && req.render_mode == bitblt_mode && rga2_service.dev_mode == 1) {
 				memcpy(&req_first, &req, sizeof(struct rga2_req));
 				if ((req_first.src.act_w != req_first.dst.act_w)
 						|| (req_first.src.act_h != req_first.dst.act_h)) {
@@ -1895,7 +1895,7 @@ static long compat_rga_ioctl(struct file *file, uint32_t cmd, unsigned long arg)
 			}
 			RGA_MSG_2_RGA2_MSG_32(&req_rga, &req);
 
-			if (first_RGA2_proc == 0 && req.bitblt_mode == bitblt_mode && rga2_service.dev_mode == 1) {
+			if (first_RGA2_proc == 0 && req.render_mode == bitblt_mode && rga2_service.dev_mode == 1) {
 				memcpy(&req_first, &req, sizeof(struct rga2_req));
 				if ((req_first.src.act_w != req_first.dst.act_w)
 						|| (req_first.src.act_h != req_first.dst.act_h)) {
-- 
2.35.3

